### PR TITLE
Dashboard screen performance fix

### LINKED ISSUE #5465: Dashboard screen performance issue
While working on #5464, I discovered a performance issue on the [dashboard screen](https://readthedocs.org/dashboard/).

Getting the latest build status and the number of builds for each project results in an extra SQL query each. This results in O(n) SQL queries where n is the number of projects displayed. It should be possible to get the latest build and number of builds for each project using a [Django annotation](https://docs.djangoproject.com/en/1.11/ref/models/querysets/#django.db.models.query.QuerySet.annotate) or a similar feature.

Screen Shot 2019-03-15 at 10 35 18 AM

### PR DESCRIPTION
**Pull request by saadmk11**
closes #5465

### CODE REVIEW

File: readthedocs/projects/views/private.py [hunk 1]
```diff
@@ -92,7 +93,11 @@ def validate_primary_email(self, user):
             notification.send()
 
     def get_queryset(self):
-        return Project.objects.dashboard(self.request.user)
+        builds = Build.objects.filter(
+            project=OuterRef('pk'), type='html', state='finished')
+        sub_query = Subquery(builds.values('success')[:1])
+        return Project.objects.dashboard(self.request.user).annotate(
+            Count('builds'), build_success=sub_query)
```
Comment (safwanrahman) [inline readthedocs/projects/views/private.py]: I think the name of the parameter should be `latest_build_success` rathar than `build_success` as its giving the latest build status.

Comment (safwanrahman) [reply readthedocs/projects/views/private.py]: what about having the `Count('builds')` also in parameter like `builds_count=Count('builds')`

File: readthedocs/projects/views/private.py [hunk 2]
```diff
@@ -92,7 +93,11 @@ def validate_primary_email(self, user):
             notification.send()
 
     def get_queryset(self):
-        return Project.objects.dashboard(self.request.user)
+        builds = Build.objects.filter(
```
Comment (safwanrahman) [inline readthedocs/projects/views/private.py]: Can you add some comment here?

File: readthedocs/projects/views/private.py (hunk 1 above)
Comment (saadmk11) [reply readthedocs/projects/views/private.py]: sure. I'll update the PR accordingly

File: readthedocs/projects/views/private.py (hunk 2 above)
Comment (saadmk11) [reply readthedocs/projects/views/private.py]: okay

**Comment by ericholscher** [comment]
Thanks! 👍

**Pull request merged**